### PR TITLE
Add 'kubectl-kadalu' plugin

### PR DESCRIPTION
**Pull request by vatsa287**
Hello `krew-index` team, we from [kadalu](kadalu.io) would like to add our cli-tool 'kubectl-kadalu'
to 'krew-index' repo. Please review and do request changes if any. Thanks! 

> kubectl-kadalu plugin helps to install Kadalu Operator &
> provides sub-commands to manage/monitor Kadalu Storage pods

Signed-off-by: Shree Vatsa N <vatsa@kadalu.io>

### CODE REVIEW

**Comment by vatsa287** [comment]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> memo **Please follow instructions at https://git.k8s.io/community/CLA.md#the-contributor-license-agreement to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment and we'll verify. Thanks.
> 
>     * If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address.  Check your existing CLA data and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
>     * If you signed the CLA as a corporation, please sign in with your organization's credentials at https://identity.linuxfoundation.org/projects/cncf to be authorized.
> 
>     * If you have done the above and are still having issues with the CLA being reported as unsigned, please log a ticket with the Linux Foundation Helpdesk: https://support.linuxfoundation.org/
> 
>     * Should you encounter any issues with the Linux Foundation Helpdesk, send a message to the backup e-mail support address at: [login-issues@jira.linuxfoundation.org](mailto:login-issues@jira.linuxfoundation.org)

I signed it.

**Comment by vatsa287** [comment]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> memo **Please follow instructions at https://git.k8s.io/community/CLA.md#the-contributor-license-agreement to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment and we'll verify. Thanks.
> 
>     * If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address.  Check your existing CLA data and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
>     * If you signed the CLA as a corporation, please sign in with your organization's credentials at https://identity.linuxfoundation.org/projects/cncf to be authorized.
> 
>     * If you have done the above and are still having issues with the CLA being reported as unsigned, please log a ticket with the Linux Foundation Helpdesk: https://support.linuxfoundation.org/
> 
>     * Should you encounter any issues with the Linux Foundation Helpdesk, send a message to the backup e-mail support address at: [login-issues@jira.linuxfoundation.org](mailto:login-issues@jira.linuxfoundation.org)

I signed it.

File: plugins/kadalu.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+apiVersion: krew.googlecontainertools.github.com/v1alpha2
+kind: Plugin
+metadata:
+  name: kadalu
+spec:
+  version: "v0.8.7"
+  homepage: https://github.com/kadalu/kadalu
```
Comment (ahmetb) [inline plugins/kadalu.yaml:7]: do you import auth plugins in your program? 
https://krew.sigs.k8s.io/docs/developer-guide/develop/best-practices/#:~:text=in%20your%20plugins.-,Import%20authentication%20plugins%20(Go),-By%20default%2C%20plugins

File: plugins/kadalu.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  shortDescription: "Deploy and manage Kadalu Operator, CSI and Storage pods"
```
Comment (ahmetb) [inline plugins/kadalu.yaml:8]: this is good but it's about 5 chars long than it fits to the "krew search" screen. Maybe delete "Deploy and"

**Review by ahmetb — CHANGES_REQUESTED** [review]
thanks for you submission

File: plugins/kadalu.yaml (hunk 1 above)
Comment (vatsa287) [reply plugins/kadalu.yaml:7]: Nope

File: plugins/kadalu.yaml (hunk 2 above)
Comment (vatsa287) [reply plugins/kadalu.yaml]: Done!

File: plugins/kadalu.yaml (hunk 1 above)
Comment (ahmetb) [reply plugins/kadalu.yaml:7]: @vatsa287 we highly recommend importing that package as it severely excludes your user base from clusters running on managed-Kubernetes services from cloud providers.

**Comment by ahmetb** [comment]
@vatsa287 would you like to wait until you make a new release with auth package imports? We can merge as is, but we ask all plugins to follow suit as well. It would improve your user experience.

**Comment by aravindavk** [comment]
> @vatsa287 would you like to wait until you make a new release with auth package imports? We can merge as is, but we ask all plugins to follow suit as well. It would improve your user experience.

`kubectl-kadalu` is a Python based kubectl extension that runs kubectl commands internally to provide the enhanced user experience for Kadalu Storage users. For example, generating the required yaml file and calling kubectl create command or running list of Pods and showing dashboard by getting information from Kadalu Pods by running kubectl exec command etc. If a user is not having access to run the kubectl command then `kubectl kadalu` will fail. Authentication is managed by the `kubectl` command itself.

**Comment by ahmetb** [comment]
Understood. I didn't realize it was a Python based plugin. 

/lgtm
/approve

Also, please consider setting up [Krew release automation](https://krew.sigs.k8s.io/docs/developer-guide/release/automating-updates/) which helps you skip manually making updates to your Krew manifests on each new version and send a pull request. It’s a GitHub workflow bot that can run every time you push a new tag. Those PRs are automatically approved.

**Pull request merged**